### PR TITLE
Use the method of a base if it's unset

### PR DESCRIPTION
**Pull request by neithernut**
After applying this PR, if you set a callback to `NULL`, the system will search for implementation i base classes.
Deinit and some other special methods are not affected by this.

### CODE REVIEW

File: src/compositor/buffer.c [hunk 1]
```diff
@@ -71,8 +71,13 @@ ws_buffer_data(
     struct ws_buffer* self
 ) {
     ws_buffer_type_id* type = (ws_buffer_type_id*) &self->obj.id;
-    if (!(type && type->get_data)) {
-        return NULL;
+
+    // search for an implementation in the base classes
+    while (!type->get_data) {
+        if (type == &WS_OBJECT_TYPE_ID_BUFFER) {
```
Comment (matthiasbeyer) [inline src/compositor/buffer.c:78]: You should add a note why we stop on the `WS_OBJECT_TYPE_ID_BUFFER` type identification! I mean, it is clear now, but maybe not later.

Comment (matthiasbeyer) [inline src/compositor/buffer.c:78]: As well for the other occurences, of course.

**Comment by neithernut** [comment]
Do I have a GO for squashing?

**Comment by matthiasbeyer** [comment]
From me: yes.

**Comment by neithernut** [comment]
Done.

**Pull request merged**